python/pecos/qeclib/steane/steane_class.py

```diff
@@ -218,40 +197,31 @@ def nonft_prep_t_plus_state(self):
 
     def prep_t_plus_state(self, reject: Bit, rus_limit: int | None = None):
         """Prepare logical T|+X> in a fault tolerant manner."""
-
-        if rus_limit is None:
-            rus_limit = self.default_rus_limit
-
-        block = Block(
+        return Block(
             self.scratch.set(0),
             PrepEncodeTPlusFTRUS(
                 d=self.d,
                 a=self.a,
                 out=self.scratch,
-                reject=self.scratch[2],
+                reject=reject or self.scratch[2],
                 flag_x=self.flag_x,
                 flag_z=self.flag_z,
                 flags=self.flags,
                 last_raw_syn_x=self.last_raw_syn_x,
                 last_raw_syn_z=self.last_raw_syn_z,
-                limit=rus_limit,
+                limit=rus_limit or self.default_rus_limit,
             ),
         )
-        if reject is not None:
-            block.extend(reject.set(self.log))
```

Comment:
This is an example of the `reject` bit being set incorrectly.  It is always set to `self.log`, even though `self.log` is unused and does not change in this method.

Reply:
Yeah, that is obviously wrong.